src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -105,7 +142,49 @@ private static Task WriteToResponseAsync<T>(T? value, HttpContext httpContext, J
+                (true, {{RequestDelegateCreationLogging.RequiredParameterNotProvidedEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.RequiredParameterNotProvidedExceptionMessage, true)}},
+                (true, {{RequestDelegateCreationLogging.ImplicitBodyNotProvidedEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.ImplicitBodyNotProvidedExceptionMessage, true)}},
+                (true, {{RequestDelegateCreationLogging.UnexpectedJsonContentTypeEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.UnexpectedJsonContentTypeExceptionMessage, true)}},
+                (true, {{RequestDelegateCreationLogging.UnexpectedFormContentTypeEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.UnexpectedFormContentTypeExceptionMessage, true)}},
+                (true, {{RequestDelegateCreationLogging.InvalidFormRequestBodyEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.InvalidFormRequestBodyExceptionMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.RequestBodyIOExceptionEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.RequestBodyIOExceptionMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.InvalidJsonRequestBodyEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.InvalidJsonRequestBodyExceptionMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.ParameterBindingFailedEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.ParameterBindingFailedLogMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.RequiredParameterNotProvidedEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.RequiredParameterNotProvidedLogMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.ImplicitBodyNotProvidedEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.ImplicitBodyNotProvidedLogMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.UnexpectedJsonContentTypeEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.UnexpectedJsonContentTypeLogMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.UnexpectedFormContentTypeEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.UnexpectedFormContentTypeLogMessage, true)}},
+                (false, {{RequestDelegateCreationLogging.InvalidFormRequestBodyEventId}}) => {{SymbolDisplay.FormatLiteral(RequestDelegateCreationLogging.InvalidFormRequestBodyLogMessage, true)}},
+                _ => throw new UnreachableException("Encountered unsupported logging EventId.")
+            };
+            void logOrThrowException(int eventId, string eventName, Exception? exception = null, int statusCode = StatusCodes.Status400BadRequest, params object?[] messageFormatArgs)
```

Comment:
`params object?[]` that's an allocation 😢

Reply:
Fixed. `params object?[]` ➡️ `params string?[[]`

Reply:
That's not the allocation I was thinking of. The array part of `params blah[]` is an allocation.
But that is a good point about the difference between this code and what RDF uses. RDF uses LoggerMessage.Define which is generic so it can avoid boxing of arguments. Although this now forces a string allocation for anything passed in. e.g. StatusCodes.Status400BadRequest when ThrowOnBadRequest is false and log level is not debug.

I think the PR is fine for now, but we should probably file an issue to consider using LoggerMessage.Define and generics to avoid the boxing and array allocations.